packages/x-data-grid/src/components/GridShadowScrollArea.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { styled, keyframes } from '@mui/system';
+
+export interface GridShadowScrollAreaProps extends React.HTMLAttributes<HTMLDivElement> {
+  children: React.ReactNode;
+}
+
+const reveal = keyframes({ from: { opacity: 0 }, to: { opacity: 1 } });
+const detectScroll = keyframes({ 'from, to': { '--scrollable': '" "' } });
+
+const ShadowScrollArea = styled('div', {
+  name: 'MuiDataGrid',
+  slot: 'ShadowScrollArea',
+})({
+  flex: 1,
+  display: 'flex',
+  flexDirection: 'column',
+  animation: detectScroll,
+  animationTimeline: '--scroll-timeline',
+  animationFillMode: 'none',
+  boxSizing: 'border-box',
+  overflow: 'auto',
+  scrollTimeline: '--scroll-timeline block',
+  '&::before, &::after': {
+    content: '""',
+    flexShrink: 0,
+    display: 'block',
+    position: 'sticky',
+    left: 0,
+    width: '100%',
+    height: '4px',
+    animation: `${reveal} linear both`,
+    animationTimeline: '--scroll-timeline',
+
+    // Custom property toggle trick:
+    // - Detects if the element is scrollable
+    // - https://css-tricks.com/the-css-custom-property-toggle-trick/
+    '--visibility-scrollable': 'var(--scrollable) visible',
+    '--visibility-not-scrollable': 'hidden',
+    visibility: 'var(--visibility-scrollable, var(--visibility-not-scrollable))',
+  },
+  '&::before': {
+    top: 0,
+    background: 'linear-gradient(to bottom, rgba(0,0,0,0.05) 0, transparent 100%)',
+    animationRange: '0 4px',
+  },
+  '&::after': {
+    bottom: 0,
+    background: 'linear-gradient(to top, rgba(0,0,0,0.05) 0, transparent 100%)',
```

Comment:
I noticed the scroll shadow doesn't show up on Firefox.

https://github.com/user-attachments/assets/d6d4be5b-61b1-47ca-a2a5-de34aa3d8887

Reply:
See https://github.com/mui/mui-x/pull/16630#discussion_r1958525744

Reply:
Cool 👌